src/Symfony/Component/HttpKernel/HttpCache/Esi.php

```diff
@@ -162,8 +162,9 @@ public function process(Request $request, Response $response)
 
         // we don't use a proper XML parser here as we can have ESI tags in a plain text response
         $content = $response->getContent();
-        $content = preg_replace('#<esi\:remove>.*?</esi\:remove>#', '', $content);
-        $content = preg_replace('#<esi\:comment[^>]*(?:/|</esi\:comment)>#', '', $content);
+        $content = preg_replace('#<esi\:remove>.*?</esi\:remove>#s', '', $content);
+        $content = preg_replace('#<esi\:comment[^>]+>#s', '', $content);
+        $content = preg_replace('#<esi\:comment[^>]*>(.+?)</esi\:comment>#s', '', $content);
```

Comment:
But the original regex doesn't pass for `<esi:comment>This is removed</esi:comment>` ;) (which is is suggested by the way it looks).

Will remove the `<esi\:comment[^>]*>(.+?)</esi\:comment>` regex then :+1: